pkg/operator/bootstrapteardown/bootstrap_teardown_controller.go

```diff
@@ -23,64 +27,65 @@ import (
 )
 
 const (
-	workQueueKey  = "key"
-	configMapName = "config"
-	configMapKey  = "config.yaml"
+	workQueueKey              = "key"
+	configMapName             = "config"
+	configMapKey              = "config.yaml"
+	conditionBootstrapRemoved = "BootstrapRemoved"
 )
 
 type BootstrapTeardownController struct {
-	operatorConfigClient        v1helpers.OperatorClient
-	clusterMemberShipController *clustermembercontroller.ClusterMemberController
+	operatorClient v1helpers.OperatorClient
 
-	etcdOperatorLister  operatorv1listers.EtcdLister
 	kubeAPIServerLister operatorv1listers.KubeAPIServerLister
 	configMapLister     corev1listers.ConfigMapLister
+	endpointLister      corev1listers.EndpointsLister
 
 	cachesToSync  []cache.InformerSynced
 	queue         workqueue.RateLimitingInterface
 	eventRecorder events.Recorder
 }
 
-// TODO wire a triggering lister
 func NewBootstrapTeardownController(
-	operatorConfigClient v1helpers.OperatorClient,
-	kubeInformersForNamespaces operatorv1helpers.KubeInformersForNamespaces,
-	clusterMemberShipController *clustermembercontroller.ClusterMemberController,
+	operatorClient v1helpers.OperatorClient,
 
+	kubeInformersForNamespaces operatorv1helpers.KubeInformersForNamespaces,
 	operatorInformers operatorv1informers.SharedInformerFactory,
 
 	eventRecorder events.Recorder,
 ) *BootstrapTeardownController {
 	openshiftKubeAPIServerNamespacedInformers := kubeInformersForNamespaces.InformersFor("openshift-kube-apiserver")
 	c := &BootstrapTeardownController{
-		operatorConfigClient:        operatorConfigClient,
-		clusterMemberShipController: clusterMemberShipController,
+		operatorClient: operatorClient,
 
-		etcdOperatorLister:  operatorInformers.Operator().V1().Etcds().Lister(),
 		kubeAPIServerLister: operatorInformers.Operator().V1().KubeAPIServers().Lister(),
 		configMapLister:     openshiftKubeAPIServerNamespacedInformers.Core().V1().ConfigMaps().Lister(),
+		endpointLister:      kubeInformersForNamespaces.InformersFor(operatorclient.TargetNamespace).Core().V1().Endpoints().Lister(),
 
 		cachesToSync: []cache.InformerSynced{
-			operatorConfigClient.Informer().HasSynced,
+			operatorClient.Informer().HasSynced,
 			operatorInformers.Operator().V1().Etcds().Informer().HasSynced,
 			operatorInformers.Operator().V1().KubeAPIServers().Informer().HasSynced,
 			openshiftKubeAPIServerNamespacedInformers.Core().V1().ConfigMaps().Informer().HasSynced,
+			openshiftKubeAPIServerNamespacedInformers.Core().V1().Endpoints().Informer().HasSynced,
+			kubeInformersForNamespaces.InformersFor(operatorclient.TargetNamespace).Core().V1().Endpoints().Informer().HasSynced,
```

Comment:
out of curiosity, why don't we have informers for pods registered in TargetNamespace?